### PR TITLE
Expose Mongo sequential ID generator as Spring bean

### PR DESCRIPTION
**Pull request by emilssolmanis**
Depended on by https://github.com/atlasapi/atlas/pull/721

### CODE REVIEW

File: src/main/java/org/atlasapi/persistence/MongoContentPersistenceModule.java [hunk 1]
```diff
@@ -217,7 +218,11 @@ public MongoContentPersistenceModule(Mongo mongo, DatabasedMongo db, MessagingMo
         return new MongoLookupEntryStore(db.collection("lookup"), 
                 persistenceAuditLog(), readPreference);
     }
-    
+
+    public @Primary @Bean IdGenerator idGenerator() {
```
Comment (JamieAP) [inline src/main/java/org/atlasapi/persistence/MongoContentPersistenceModule.java]: contentIdGenerator()

File: src/main/java/org/atlasapi/persistence/ContentPersistenceModule.java [hunk 2]
```diff
@@ -41,4 +43,6 @@
 	ProductResolver productResolver();
 	
 	PeopleQueryResolver peopleQueryResolver();
+
+	IdGenerator idGenerator();
```
Comment (JamieAP) [inline src/main/java/org/atlasapi/persistence/ContentPersistenceModule.java]: contentIdGenerator()

**Comment by Dalamar42** [comment]
I'd check if there are places where the content id generator is being used where it might be worth replacing it with this instance to avoid duplication. It might not be possible to do this cleanly in which case this is fine. Other than that looks good to me.

**Comment by emilssolmanis** [comment]
@Dalamar42 good spot, was only 1 place though.

**Pull request merged**